### PR TITLE
[Fix #48] Fix RegexpMatch false positive

### LINKED ISSUE #48: False positives for `match` in RegexpMatch
The `RegexpMatch` is a bit too eager in flagging usage of `match`. Here's a simple example that is flagged when extending Rubocop's own API:

```ruby
class MyCop < Cop
  KERNEL_PATTERN = RuboCop::NodePattern.new('(const nil? :Kernel)')
  def on_send(node)
    if KERNEL_PATTERN.match(node)
      # do stuff
    end
  end
end
```

I would suggest making the `match_method?` logic more selective, and only flag usage by receiver types known to have a regexp matching `match` method.  At risk of getting too into implementation details, rather than:
```ruby
def_node_matcher :match_method?, <<-PATTERN
  {
    (send _recv :match _)
    (send _recv :match _ (int ...))
  }
PATTERN
```
`RegexpMatch` could instead do:
```ruby
def_node_matcher :match_method?, <<-PATTERN
  {
    (send {regexp str sym} :match _)
    (send {regexp str sym} :match _ (int ...))
  }
PATTERN
```
I'm happy to open a PR if this is agreeable.
--------

## Expected behavior

The snippet above should not trigger a `Performance/RegexpMatch` violation

## Actual behavior

```
Offenses:

repro.rb:4:8: C: Performance/RegexpMatch: Use match? instead of match when MatchData is not used.
    if KERNEL_PATTERN.match(node)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

`rubocop file.rb --only Performance/RegexpMatch` with the above snippet saved in `file.rb`

## RuboCop version

```
0.68.0 (using Parser 2.6.0.0, running on ruby 2.4.5 x86_64-darwin17)
```

### PR DESCRIPTION
**Pull request by dduugg**
This fixes false positives in `Performance/RegexpMatch` by only flagging single-argument `match` methods where either the receiver or the argument is a literal of type `Regexp`, `String`, `Symbol`.

Resolves https://github.com/rubocop-hq/rubocop-performance/issues/48

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-performance/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-performance/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
+### Bug fixes
+
+* [#48](https://github.com/rubocop-hq/rubocop-performance/issues/48): Fix false positives in Performance/RegexpMatch. ([@dduugg][])
+
 ## 1.2.0 (2019-05-04)
 
 ### Bug fixes
 
-* [#47](https://github.com/rubocop-hq/rubocop-performance/pull/47): Fix a false negartive for `Performance/RegexpMatch` when using RuboCop 0.68 or higher. ([@koic][])
+* [#47](https://github.com/rubocop-hq/rubocop-performance/pull/47): Fix a false negative for `Performance/RegexpMatch` when using RuboCop 0.68 or higher. ([@koic][])
```
Comment (dduugg) [inline CHANGELOG.md]: unrelated typo fix

File: lib/rubocop/cop/performance/regexp_match.rb [hunk 2]
```diff
@@ -80,14 +80,14 @@ class RegexpMatch < Cop
         # Constants are included in this list because it is unlikely that
         # someone will store `nil` as a constant and then use it for comparison
         TYPES_IMPLEMENTING_MATCH = %i[const regexp str sym].freeze
-        MSG =
-          'Use `match?` instead of `%<current>s` when `MatchData` ' \
+        MSG = 'Use `match?` instead of `%<current>s` when `MatchData` ' \
```
Comment (dduugg) [inline lib/rubocop/cop/performance/regexp_match.rb]: This is to resolve a `Metrics/ClassLength` cop violation.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,11 +2,15 @@
 
 ## master (unreleased)
 
+### Bug fixes
+
+* [#48](https://github.com/rubocop-hq/rubocop-performance/issues/48): Fix false positives in `Performance/RegexpMatch`. ([@dduugg][])
```
Comment (koic) [inline CHANGELOG.md]: Could you write a little more about what false positives have been fixed?

**Comment by koic** [comment]
This code looks good. Could you please rebase using latest master branch? A warning will probably occur because Ruby 2.2 was dropped.

**Comment by dduugg** [comment]
@koic I've updated the changelog to detail the change, and rebased on current master. Please take another look. Thanks!

**Comment by koic** [comment]
Thanks!
